runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java

```diff
@@ -689,6 +689,15 @@ else if (target.equals("teleport to house"))
 			{
 				swapTeleport(target, option, "outside", index);
 			}
+			else if (target.equals("npc contact"))
+			{
+				MenuEntry[] entries = client.getMenuEntries();
+				if (entries != null && entries.length > 1)
+				{
+					String npcName = Text.removeTags(entries[entries.length - 2].getOption()).toLowerCase();
```

Comment:
This looks unusual and should probably be done similarly to the jewellery box's recent destination swaps instead.